src/main/steps/applicant/statement-of-service/choose-parties/content.ts

```diff
@@ -0,0 +1,210 @@
+let updatedForm: FormContent;
+
+const updateFormFields = (form: FormContent, formFields: FormContent['fields']): FormContent => {
+  updatedForm = {
+    ...form,
+    fields: {
+      ...formFields,
+      ...(form.fields ?? {}),
+    },
+  };
+
+  return updatedForm;
+};
+
+export const generateFormFields = (caseData: Partial<CaseWithId>): GenerateDynamicFormFields => {
+  const errors = {
+    en: {},
+    cy: {},
+  };
+
+  const fields = {
+    sos_partiesServed: {
+      type: 'checkboxes',
+      label: l => l.whowasserved,
+      values: getParties(caseData).map(respondent => ({
+        name: 'sos_partiesServed',
+        value: respondent.id,
+        label: respondent.value,
+        selected: caseData.sos_partiesServed?.includes(respondent.id) || false,
+      })),
+      validator: atLeastOneFieldIsChecked,
+    },
+    sos_partiesServedDate: {
+      type: 'date',
+      classes: 'govuk-date-input',
+      label: l => l.servedDate,
+      labelSize: 's',
+      hint: l => l.servedDateHint,
+      values: [
+        {
+          label: l => l.dateFormat['day'],
+          name: 'day',
+          classes: 'govuk-input--width-2',
+          attributes: { maxLength: 2, pattern: '[0-9]*', inputMode: 'numeric' },
+          value: _.get(caseData, 'sos_partiesServedDate.day', undefined),
+        },
+        {
+          label: l => l.dateFormat['month'],
+          name: 'month',
+          classes: 'govuk-input--width-2',
+          attributes: { maxLength: 2, pattern: '[0-9]*', inputMode: 'numeric' },
+          value: _.get(caseData, 'sos_partiesServedDate.month', undefined),
+        },
+        {
+          label: l => l.dateFormat['year'],
+          name: 'year',
+          classes: 'govuk-input--width-4',
+          attributes: { maxLength: 4, pattern: '[0-9]*', inputMode: 'numeric' },
+          value: _.get(caseData, 'sos_partiesServedDate.year', undefined),
+        },
+      ],
+      parser: body => covertToDateObject('sos_partiesServedDate', body as Record<string, unknown>),
+      validator: value =>
+        areDateFieldsFilledIn(value as CaseDate) ||
+        isDateInputInvalid(value as CaseDate) ||
+        isFutureDate(value as CaseDate),
+    },
+  };
+
+  return { fields, errors };
+};
+
+export const form: FormContent = {
+  fields: {},
+  onlyContinue: {
+    text: l => l.onlycontinue,
+  },
+};
+
+const getParties = (userCase: Partial<CaseWithId>) => {
+  const parties: { id: string; value: string }[] = [];
+  userCase?.respondents?.forEach(respondent => {
+    if (respondent.value.response.citizenFlags?.isApplicationToBeServed !== YesOrNo.YES) {
+      parties.push({
+        id: respondent.id,
+        value: respondent.value.firstName + ' ' + respondent.value.lastName,
+      });
+    }
+  });
+  return parties;
+};
+
+export const generateContent: TranslationFn = content => {
+  const translations = languages[content.language];
+  const url = `?context=${content.additionalData?.req.params.context}&isSos=Yes`;
```

Comment:
why is url required to be constructed with query params. I have already said, query param will not work with language toggle. use params instead. In this case url itself is not required to be constructed.

Reply:
For upload scenario, in order for params to be present as part of request from upload action we need this.